docs/beginners-guide-caffe2dml.md

```diff
@@ -58,67 +48,167 @@ specifies to Caffe2DML to use following configuration when generating the traini
 - `display: 100`: Display training loss after every 100 iterations.
 - `test_interval: 500`: Display validation loss after every 500 iterations.
 - `test_iter: 10`: Validation data size = 10 * BATCH_SIZE.
- 
+
+```python
+from systemml.mllearn import Caffe2DML
+import urllib
+
+# Download the Lenet network
+urllib.urlretrieve('https://raw.githubusercontent.com/apache/systemml/master/scripts/nn/examples/caffe2dml/models/mnist_lenet/lenet.proto', 'lenet.proto')
+urllib.urlretrieve('https://raw.githubusercontent.com/apache/systemml/master/scripts/nn/examples/caffe2dml/models/mnist_lenet/lenet_solver.proto', 'lenet_solver.proto')
```

Comment:
One issue here is that Python3 moved `urlretrieve` into a new `urllib.request` package.  To make sure this demo works on Python 2 & 3, we could use the following:

```
from systemml.mllearn import Caffe2DML
try:
  from urllib.request import urlretrieve  # py3
except:
  from urllib import urlretrieve  # py2

# Download the Lenet network
urlretrieve('https://raw.githubusercontent.com/apache/systemml/master/scripts/nn/examples/caffe2dml/models/mnist_lenet/lenet.proto', 'lenet.proto')
urlretrieve('https://raw.githubusercontent.com/apache/systemml/master/scripts/nn/examples/caffe2dml/models/mnist_lenet/lenet_solver.proto', 'lenet_solver.proto')
```